File: c#/vs/Common7/IDE/ProjectTemplatesCache/CSharp/Web/2052/AjaxServerControl/ServerControl1.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
$if$ ($targetframeworkversion$ >= 3.5)using System.Xml.Linq;
$endif$
namespace $safeprojectname$
{
    /// <summary>
    /// ServerControl1 的摘要说明
    /// </summary>
    public class ServerControl1 : ScriptControl
    {
        public ServerControl1()
        {
            //
            // TODO: 在此处添加构造函数逻辑
            //
        }
        protected override IEnumerable<ScriptDescriptor>
                GetScriptDescriptors()
        {
            ScriptControlDescriptor descriptor = new ScriptControlDescriptor("$safeprojectname$.ClientControl1", this.ClientID);
            yield return descriptor;
        }

        // 生成脚本引用
        protected override IEnumerable<ScriptReference>
                GetScriptReferences()
        {
            yield return new ScriptReference("$safeprojectname$.ClientControl1.js", this.GetType().Assembly.FullName);
        }
    }
}

File: c#/vs/Common7/IDE/ProjectTemplatesCache/VisualBasic/Web/2052/AjaxServerControl/ServerControl1.vb
Public Class ServerControl1
    Inherits ScriptControl

    Protected Overrides Function GetScriptDescriptors() As IEnumerable(Of ScriptDescriptor)
        Dim descriptor As New ScriptControlDescriptor("$safeprojectname$.ClientControl1", Me.ClientID)

        Return New List(Of ScriptDescriptor) From {descriptor}
    End Function

    ' 生成脚本引用
    Protected Overrides Function GetScriptReferences() As IEnumerable(Of ScriptReference)
        Dim scriptRef As New ScriptReference("$safeprojectname$.ClientControl1.js", Me.GetType().Assembly.FullName)

        Return New List(Of ScriptReference) From {scriptRef}
    End Function
End Class
